pkg/debuginfo/manager.go

```diff
@@ -123,25 +128,30 @@ func (di *Manager) removeAsUploading(buildID string) {
 
 // EnsureUploaded ensures that the extracted or the found debuginfo for the given buildID is uploaded.
 func (di *Manager) EnsureUploaded(ctx context.Context, objFiles []*objectfile.MappedObjectFile) {
-	_, err, _ := di.sfg.Do("ensure-uploaded", func() (interface{}, error) {
-		g := errgroup.Group{}
-		g.SetLimit(4) // Arbitrary limit.
-		for _, objFile := range objFiles {
-			logger := log.With(di.logger, "buildid", objFile.BuildID, "path", objFile.Path)
-			objFile := objFile
-			g.Go(func() error {
-				err := di.ensureUploaded(ctx, objFile)
-				if err != nil {
-					level.Error(logger).Log("msg", "failed to ensure debuginfo is uploaded", "err", err)
-				}
-				return nil
-			})
+	go func() {
+		_, err, _ := di.sfg.Do("ensure-uploaded", func() (interface{}, error) {
+			g := errgroup.Group{} // errgroup.WithContext doesn't work for this use case, we want to continue uploading even if one fails.
```

Comment:
Should we use [`WithContext`](https://cs.opensource.google/go/x/sync/+/refs/tags/v0.1.0:errgroup/errgroup.go;l=45;bpv=1;bpt=1)?

Reply:
I have added the comment for this question 😅

Reply:
But if the context gets canceled from the caller, we already propagate it down, so the grpc calls will be canceled. And group will exit.

Reply:
> I have added the comment for this question 😅

Oops, my bad, didn't see it